odoo/addons/base/tests/test_translate.py

```diff
@@ -1108,6 +1108,34 @@ def test_sync_xml_close_terms(self):
         self.assertEqual(view.with_env(env_en).arch_db, archf % terms_en)
         self.assertEqual(view.with_env(env_fr).arch_db, archf % ('RandomRandom1', 'SomethingElse', 'AléatoireAléatoire3'))
 
+    def test_sync_xml_upgrade(self):
+         archf = '<form>%s<div>%s</div></form>'
+         terms_en = ('Draft', '<span invisible="1">Draft</span>')
+         terms_fr = ('Brouillon', '<span invisible="1">Brouillon</span>')
+         view = self.create_view(archf, terms_en, en_US=terms_en, fr_FR=terms_fr)
+
+         archf = '<form><div>%s</div></form>'
+         terms_en = ('<span invisible="0">Draft</span>')
+         terms_fr = ('<span invisible="0">Draft</span>')  # translations will be updated when import translations from po files
```

Comment:
we cannot do what we do in 17.0 in the translate.term_adapter is not available in 16.0 which can convert old translation
`'<span invisible="1">Brouillon</span>'`
to
`'<span invisible="0">Brouillon</span>'`
according to the new matched term
`'<span invisible="0">Draft</span>'`

So in 16.0, we must drop the translation and relay on the latest translations to update the translation

Reply:
Based on https://github.com/odoo/odoo/pull/194181

Wouldn't it be better to backport the term adapter to 16? Originally the term adapter was _strongly_ needed in 17 because of the switch from domains to python expression in modifiers. Now, as the linked PR describes, it looks like the adapter is for all attributes to avoid upgrade issues.